### PR TITLE
AP-2315 Migrate package_build stage to k8s runners

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

Migrate `package_build` stage to run on k8s runners.
Fix a problem where `make -j workers` would use all the CPU of the host on k8s runners. Because ohai["cpu"] detects CPUs of the host.

Why did I set the memory request/limit to 6/12:

image

### Motivation

Complete migration to k8s gitlab runners

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by chouquette — APPROVED** [review]

File: tasks/agent.py [hunk 1]
```diff
@@ -525,6 +525,9 @@ def get_omnibus_env(
         env['SYSTEM_PROBE_BIN'] = system_probe_bin
     env['AGENT_FLAVOR'] = flavor.name
 
+    if os.environ.get('KUBERNETES_CPU_REQUEST'):
+        env['OMNIBUS_WORKERS_OVERRIDE'] = str(int(os.environ.get('KUBERNETES_CPU_REQUEST')) + 1)
+
```
Comment (KSerrania) [inline tasks/agent.py:530]: Could you add a comment explaining why we're adding this here?

File: omnibus/omnibus.rb [hunk 2]
```diff
@@ -14,6 +14,10 @@
 # Do not set this environment variable if building locally.
 # This cache is only necessary because Datadog is building
 # the agent over and over again in a highly distributed environment.
```
Comment (KSerrania) [inline omnibus/omnibus.rb:16]: Could you move this comment (which refers to `S3_OMNIBUS_CACHE_BUCKET`) below the code block you added?

File: .gitlab/package_build/deb.yml [hunk 3]
```diff
@@ -64,6 +64,10 @@
     - $S3_CP_CMD $OMNIBUS_PACKAGE_DIR/datadog${FLAVOR:+-$FLAVOR}-agent_*_${PACKAGE_ARCH}.deb $S3_ARTIFACTS_URI/$DESTINATION_DEB
     - $S3_CP_CMD $OMNIBUS_PACKAGE_DIR/datadog${FLAVOR:+-$FLAVOR}-agent-dbg_*_${PACKAGE_ARCH}.deb $S3_ARTIFACTS_URI/$DESTINATION_DBG_DEB
     - !reference [.upload_sbom_artifacts]
+  variables:
+    KUBERNETES_MEMORY_REQUEST: "6Gi"
+    KUBERNETES_MEMORY_LIMIT: "12Gi"
```
Comment (KSerrania) [inline .gitlab/package_build/deb.yml:69]: Is there any particular reason why we're choosing these memory values? Have you checked how much memory jobs on your branch take on average?

**Review by KSerrania — COMMENTED** [review]
Looks good to me overall, asked some questions for clarification.

**Comment by KSerrania** [comment]
Also, have you compared the job durations to what we're seeing on `main`? Is there any significant difference?

File: .gitlab/package_build/deb.yml (hunk 3 above)
Comment (chouetz) [reply .gitlab/package_build/deb.yml:69]: any relation with the [study](https://docs.google.com/spreadsheets/d/1XmicdhhJCn6fMCPqFlLDfccR5rAJM8ed5-KE7gPNCK4/edit?usp=sharing) done few time ago?

Comment (KSerrania) [reply .gitlab/package_build/deb.yml:69]: I don't think it applies here, build jobs have different requirements (and memory usage patterns) than unit test jobs.

**Comment by KevinFairise2** [comment]
Yes I think it is a problem linked with how containers are designed. And how we can retrieve information about CPU. I found several issue about this ([example](https://github.com/kubernetes/kubernetes/issues/92904#issuecomment-658981227))

File: omnibus/omnibus.rb (hunk 2 above)
Comment (KevinFairise2) [reply omnibus/omnibus.rb:16]: Ah yes sorry for this!

**Comment by KevinFairise2** [comment]
> Also, have you compared the job durations to what we're seeing on `main`? Is there any significant difference?

I did not launch a lot of pipeline with the changes, but I could not notice any significant modification in the duration of the job

**Review by pgimalac — APPROVED** [review]

**Review by KSerrania — APPROVED** [review]

**Pull request merged**